### ISSUE TITLE
Error uninstalling: Exception has been thrown by the target of an invocation.

### ISSUE THREAD

**Issue by decrypter82** [PaaaulZ/SilentClickOnce#10]
Hi.

First at all. Good idea on developing this wonderful app. I have an issue on v1.5. I have Windows 11 Pro x64 and a few ClickOnce apps, one named Admin Module. 
I use the syntax SilentClickOnce -u "Admin Module" > AdminMod.log 
and I get this error

[2023-06-29 10:33:14] [?] PublicKeyToken: 0000000000000000, Culture: neutral, processorArchitecture: x86
[2023-06-29 10:33:14] [-] Error uninstalling: Exception has been thrown by the target of an invocation.

   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at SilentClickOnce.Form1.Uninstall(String applicationName)

I wrote the name in quotes as thats how I found in in my registry and due of the space between the words

![image](https://github.com/PaaaulZ/SilentClickOnce/assets/26398249/b06fdd22-c099-46c7-a4ec-bc553a35bab7)

IF I try AdminModule it says [2023-06-29 10:39:06] [-] Application 'AdminModule' not found in Registry.

Any idea?

**Comment by PaaaulZ** [comment]
I honestly can't figure out why, I'll try adding more logging to check if something was not found and set to Nothing

**Comment by nicebytenl** [comment]
I have the same issue when uninstalling an application. In which location is silentclickonce searching for the app name in the Windows Registry?

**Comment by PaaaulZ** [comment]
HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall

Regarding this issue it finds the name, tokens and everthing but when passing the data to the invoke call it fails

**Comment by decrypter82** [comment]
> HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall
> 
> Regarding this issue it finds the name, tokens and everthing but when passing the data to the invoke call it fails

Hi PaaaulZ

Just letting you know, this is what I got on my registry based on that registry route:

![image](https://github.com/PaaaulZ/SilentClickOnce/assets/26398249/86f28005-5305-4486-ad7a-d4bc31c2cd77)

Using this SilentClickOnce -u "Admin Module" > AdminMod.log I get that error and using SilentClickOnce -u AdminModule > AdminMod.log says app not found, and, makes sense as there is a space between the two word in the app name in the registry as shown in the picture.
But this is the whole set I have in the registry for that app in that route:

![image](https://github.com/PaaaulZ/SilentClickOnce/assets/26398249/73fafa8b-6061-4c6f-84f4-ca8258b463cf)

**Comment by PaaaulZ** [comment]
@decrypter82 can you try this new release and send me the logs? I added a logging line for the inner exception, maybe it adds something important.
I think it will say something like "the application does not exist". The application name is "Admin Module" but the ClickOnce is called "AdminModule.application" while the software tries to uninstall "Admin Module.application".

I have to add a fix but can't do it now, I will work on it later today though

**Comment by nicebytenl** [comment]
With the latest version I get the same Result: 

[2023-07-03 14:19:41] [-] Application 'Vena' not found in registry

The uninstallstring is in this location: Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\2b78967e7a690b58

**Comment by PaaaulZ** [comment]
@nicebytenl is that application present in "HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall" in one of the keys? Sounds like two different issues

**Comment by nicebytenl** [comment]
Yes! 
![image](https://github.com/PaaaulZ/SilentClickOnce/assets/137612242/8bd69ef6-3081-4dbf-bbb6-ff1d0e49d695)

**Comment by nicebytenl** [comment]
Maybe it's because the file is APPLICATIONX.appref-ms, instead of .application?

**Comment by PaaaulZ** [comment]
@nicebytenl it's because the DisplayName is Vena - 2 and is different from the real name Vena.
I will fix this too with the new patch, if you want to test right now change the value for DisplayName in the registry and set it to Vena. It should work

**Comment by nicebytenl** [comment]
Still not working, see screenshot: 

![image](https://github.com/PaaaulZ/SilentClickOnce/assets/137612242/06eac63c-7540-422a-8018-6c542d6922ec)

**Comment by decrypter82** [comment]
HI Paul

I **tried SilentClickOnce -u "Admin Module.application" > test1.log**

I get this in log:
[2023-07-03 08:27:33] [-] Application 'Admin Module.application' not found in registry

**Tried SilentClickOnce -u "AdminModule.application" > test1.log** 
Log:
[2023-07-03 08:28:49] [-] Application 'AdminModule.application' not found in registry

**Tried SilentClickOnce -u "AdminModule" > test1.log**
Log:
[2023-07-03 08:31:10] [-] Application 'AdminModule' not found in registry

**Tried this SilentClickOnce -u "Admin Module" > test1.log**
Log:

[2023-07-03 08:35:50] [?] PublicKeyToken: 0000000000000000, Culture: neutral, processorArchitecture: x86
[2023-07-03 08:35:50] [-] Error uninstalling: System.Deployment.Application.DeploymentException: Application is not installed.
   at System.Deployment.Application.SubscriptionStore.CheckInstalled(SubscriptionState subState)
   at System.Deployment.Application.SubscriptionStore.UninstallSubscription(SubscriptionState subState)

---------Exception has been thrown by the target of an invocation.

---------   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at SilentClickOnce.Form1.Uninstall(String applicationName)

Four combinations done and I got those responses in log. Seems like ClickonceApps are kinda tricky.
Done using your last release v1.5.1

**Comment by PaaaulZ** [comment]
@decrypter82 I added a new version, hoping this will work :c

**Comment by decrypter82** [comment]
Application Successfully removed.
Log
[2023-07-03 09:22:10] [?] PublicKeyToken: 0000000000000000, Culture: neutral, processorArchitecture: x86
[2023-07-03 09:22:10] [?] Uninstall string: rundll32.exe dfshim.dll,ShArpMaintain AdminModule.application, Culture=neutral, PublicKeyToken=0000000000000000, processorArchitecture=x86
[2023-07-03 09:22:10] [?] Using new method
[2023-07-03 09:22:10] [+] Succesfully uninstalled

it disappeared from Installed Apps list in Windows Settings, also it is not in the Registry anymore
I tried with other apps and worked also. I reinstalled the removed app and uninstalled again with your tool and worked still.
Congratulations. Good job for this great idea again. and thank you so much for your quick support. 
:-)
